### PR TITLE
[IMP] point_of_sale: search all products regardless of selected category

### PR DESCRIPTION
**Pull request by AH-Yussef**

### CODE REVIEW

File: addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js [hunk 1]
```diff
@@ -336,17 +336,13 @@ export class ProductScreen extends Component {
...
-
-        const exactMatches = products.filter((product) => product.exactMatch(searchWord));
+        const exactMatches = this.products.filter((product) => product.exactMatch(searchWord));
 
         if (exactMatches.length > 0 && searchWord.length > 2) {
             return exactMatches;
         }
 
-        const fuzzyMatches = fuzzyLookup(unaccent(searchWord, false), products, (product) =>
+        const fuzzyMatches = fuzzyLookup(unaccent(searchWord, false), this.products, (product) =>
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js:345]: could you check what the results of the search look like if you remove the `exactMatches` logic and do the fuzzy search directly ? The fuzzy search algorithm should give more weight to exact matches so maybe the result would be similar.  That would help in reducing some code.

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Comment by pedrambiria** [comment]
@AH-Yussef Hello,
Why did you remove the exact match? What was the initial issue? Now it's not possible to search with barcode.

**Comment by vlst-odoo** [comment]
> @AH-Yussef Hello, Why did you remove the exact match? What was the initial issue? Now it's not possible to search with barcode.

Hello @pedrambiria , 

I suggested doing this. The idea was that `fuzzyLookup` would return the exact matches first anyways. Is that not the case?

Can you add a tour here https://github.com/odoo/odoo/pull/187129/files  where you add a couple of products and then in the frontend observe which ones are shown based on the search string? It feels like every couple of weeks we have some issues with this search and this should solve the issue forever

**Pull request closed without merging**